### PR TITLE
ENH: add voronoi_polygons method

### PR DESCRIPTION
**Pull request by martinfleis**
xref #2010

### CODE REVIEW

**Review by m-richards — APPROVED** [review]
Thanks Martin, just some small comments.

I think one day it might be nice for users if `extend_to` for a geodataframe accepted a column name as input as well, but happy not to add scope to this PR.

File: geopandas/base.py [hunk 1]
```diff
@@ -773,9 +773,96 @@ def delaunay_triangles(self, tolerance=0.0, only_edges=False):
...
+        1    GEOMETRYCOLLECTION (POLYGON ((-14.99485 -17, -...
+        2    GEOMETRYCOLLECTION (POLYGON ((-15 -17, -15 22....
+        dtype: geometry
+
+        See also
+        --------
+        GeoSeries.delaunay_triangles : Delaunay triangulation around vertices
+        """
+        if isinstance(extend_to, GeoPandasBase):
+            extend_to = extend_to.align(self)[0]
```
[the remark this answers is not in the document]
Comment (martinfleis) [reply geopandas/base.py:861]: Let's pick it up as a follow-up to deal with all of it at once.

[the remark this answers is not in the document]
Comment (m-richards) [reply geopandas/base.py:922]: split out to #3256

**Comment by jorisvandenbossche** [comment]
Also here another option would be to consider all geometries of the GeoSeries as input together, and then explode the resulting collection (like you did for `polygonize` and `build_area`). Do we have some rationale of why we would do it here rather elementwise, while for the others considering the whole series?

**Comment by m-richards** [comment]
@martinfleis I'm happy to defer to you and Joris on what makes the most sense, I don't have the widest exposure to different GIS operations. I can see the argument that it's probably more likely you would want to consider all inputs together.


(Though with the current API, if you want to consider all values, you can wrap gs.array in a geometrycollection in a geoseries, to get what you want. If we change the API, and you wanted what is currently implemented (voronoi polygons per row), I don't think there's a convenient way to do this?)

**Comment by martinfleis** [comment]
> If we change the API, and you wanted what is currently implemented (voronoi polygons per row), I don't think there's a convenient way to do this?

Use shapely directly?

**Comment by m-richards** [comment]
> Use shapely directly?

Yeah that's true, I suppose it's worth keeping in mind we're optimising towards the most common use case, nothing stopping power users reaching for shapely directly

**Comment by martinfleis** [comment]
I have updated the behaviour here. Regarding `delaunay_triangles`, which should behave the same, I'll open another PR but it is worth noting that `delaunay_triangles` is included in 0.14 so this will be a hard breaking change...

**Review by jorisvandenbossche — APPROVED** [review]

**Pull request merged**